label_studio_sdk/project.py

```diff
@@ -1361,3 +1365,122 @@ def connect_azure_export_storage(
         }
         response = self.make_request('POST', '/api/storages/export/azure', json=payload)
         return response.json()
+
+    def assign_by_sampling(
+            self,
+            users: List[int],
+            assign_function: Callable,
+            view_id: int = None,
+            method: AssignmentSamplingMethod = AssignmentSamplingMethod.RANDOM,
+            fraction: float = 1.0,
+
+    ):
+        """
+        Assigning tasks to Reviewers or Annotators by assign_function with method by fraction from view_id
+        Parameters
+        ----------
+        users: List[int]
+            users' IDs list
+        assign_function: Callable
+            Function to assign tasks by list of user IDs
+        view_id: int
+            Optional, view ID to filter tasks to assign
+        method: AssignmentSamplingMethod
+            Optional, Assignment method
+        fraction
+            Optional, expresses the size of dataset to be assigned
+        Returns
+        -------
+        list[dict]
+            List of dicts with counter of created assignments
+        """
+        final_results = []
+        # Get tasks to assign
+        tasks = self.get_tasks(view_id=view_id, only_ids=True)
+        # Choice fraction of tasks
+        if fraction != 1.0:
+            k = int(len(tasks) * fraction)
+            tasks = sample(tasks, k)
+        # Check how many tasks for each user
+        n_tasks = max(int(len(tasks) / len(users)), 1)
```

Comment:
Please do this at the beginning to avoid zero division: 
```
assert len(users) > 0, 'Users are empty'
```

Reply:
Also we need to check tasks 
```
if len(tasks) == 0:
  logger.warning('No tasks found')
  return None
```

Reply:
Fixed